src/entities/inputValidator/utils/validateTokens.ts

```diff
@@ -68,3 +68,18 @@ export const validatePoolHasBpt = (poolState: PoolStateInput) => {
         );
     }
 };
+
+export const validateCreatePoolTokens = (
+    tokens: { tokenAddress: string }[],
+) => {
+    const tokenAddresses = tokens.map((t) => t.tokenAddress);
+    if (tokenAddresses.length !== new Set(tokenAddresses).size) {
+        throw new Error('Duplicate token addresses');
+    }
+    if (tokens.length > 4) {
+        throw new Error('Maximum of 4 tokens allowed');
+    }
```

Comment:
Can you please check how clear is the error returned from the SC to a request with > tokens than allowed?
If we consider it's good enough, we could simply let it fail at the SC level, right?
If we think the error message can be improved, then it would be nice to replicate the limit the SC defines (5 tokens on CSP and 8 tokens on weighted pools, right?)

Reply:
Done